CMakeLists.txt

```
@@ -134,7 +134,14 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
 endif()
 
 set(ENABLE_MPI OFF)
-set(NUM_MPI_PROC_TESTING "4" CACHE STRING "Number of mpi processors to use when running tests with MPI")
+if( NOT NUM_MPI_PROC_TESTING )
+  set(NUM_MPI_PROC_TESTING "4" CACHE STRING "Number of mpi processors to use when running tests with MPI")
+else()
+  get_property(EXISTS_IN_CACHE CACHE NUM_MPI_PROC_TESTING PROPERTY TYPE)
+  if( NOT EXISTS_IN_CACHE )
+    set(NUM_MPI_PROC_TESTING "${NUM_MPI_PROC_TESTING}" CACHE STRING "Number of mpi processors to use when running tests with MPI")
+  endif()
+endif()
```

Revert this, I think - you had just previously made a mistake in setting the cache variable in CI.cmake.

```suggestion
set(NUM_MPI_PROC_TESTING "4" CACHE STRING "Number of mpi processors to use when running tests with MPI")
```